src/client.rs

```diff
@@ -543,11 +576,17 @@ impl<T: Read + Write> Session<T> {
         S1: AsRef<str>,
         S2: AsRef<str>,
     {
-        self.run_command_and_read_response(&format!(
-            "FETCH {} {}",
-            sequence_set.as_ref(),
-            query.as_ref()
-        ))
+        let command;
+        self.run_command_and_read_response(if sequence_set.as_ref().is_empty() {
+            "NOOP"
```

Comment:
We can probably avoid contacting the server at all here by simply calling `parse_fetches` directly on an empty string.